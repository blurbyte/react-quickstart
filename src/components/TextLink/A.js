import styled from 'styled-components';

const A = styled.a`
  color: #ff4326;

  &:hover {
    text-decoration: underline;
  }
`;

export default A;
