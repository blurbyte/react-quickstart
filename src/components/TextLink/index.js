import React, { PropTypes } from 'react';
import A from './A';

const TextLink = (props) => (
  <A href={props.href} target="_blank">
    {props.children}
  </A>
);

TextLink.propTypes = {
  href: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default TextLink;
