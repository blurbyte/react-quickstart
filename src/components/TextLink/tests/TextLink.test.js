import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import TextLink from '../index';

describe('<TextLink />', () => {
  it('should render correctly', () => {
    // props init
    const props = {
      href: 'https://google.com'
    };

    // component init
    const component = renderer.create(
      <TextLink {...props}>Google.com</TextLink>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
