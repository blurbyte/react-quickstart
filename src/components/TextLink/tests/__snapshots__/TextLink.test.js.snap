// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> should render correctly 1`] = `
.ifuQuG {
  color: #ff4326;
}

.ifuQuG:hover {
  text-decoration: underline;
}

<a
  className="ifuQuG"
  href="https://google.com"
  target="_blank"
>
  Google.com
</a>
`;
